pkg/controllers/node/health/controller.go

```diff
@@ -70,6 +81,26 @@ func (c *Controller) Reconcile(ctx context.Context, node *corev1.Node) (reconcil
 		return reconcile.Result{}, nodeutils.IgnoreNodeClaimNotFoundError(err)
 	}
 
+	nodePoolName, found := nodeClaim.Labels[v1.NodePoolLabelKey]
+	if found {
```

Comment:
nit: consider adding a comment here to indicate why we are doing both of these checks